tests/components/withings/test_sensor.py

```diff
@@ -46,10 +29,14 @@ async def test_all_entities(
 ) -> None:
     """Test all entities."""
     await setup_integration(hass, polling_config_entry)
+    entity_registry = er.async_get(hass)
+    entities = er.async_entries_for_config_entry(
+        entity_registry, polling_config_entry.entry_id
+    )
 
-    for sensor in MEASUREMENT_SENSORS + SLEEP_SENSORS:
-        entity_id = await async_get_entity_id(hass, sensor.key, USER_ID, SENSOR_DOMAIN)
-        assert hass.states.get(entity_id) == snapshot
+    for entity in entities:
+        if entity.platform == Platform.SENSOR:
+            assert hass.states.get(entity.entity_id) == snapshot
```

Comment:
If some how no sensor get registered this would pass.

Maybe add an assert that there are sensor entities?

Reply:
If that would be the case, syrupy would fall iirc